### PR TITLE
*: upgrade x/exp/slices version

### PR DESCRIPTION
**Pull request by D3Hunter**
### What problem does this PR solve?

Issue Number: ref #45976

Problem Summary:

### What changed and how does it work?
in tiflow repo, we want to pick https://github.com/pingcap/tiflow/pull/11700, but it upgrades `x/exp` which is in-compatible with tidb/parser. 

this PR is a manually pick of https://github.com/pingcap/tidb/pull/53785

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: DEPS.bzl
Comment (D3Hunter) [inline DEPS.bzl:1]: copied from master

**Review by hawkingrei — APPROVED** [review]

**Review by Benjamin2037 — APPROVED** [review]
LGTM

**Comment by D3Hunter** [comment]
/hold

for a while, to confirm that every thing tiflow needed are changed

**Pull request merged**